File: src/com/gelakinetic/mlb_elo_calculator/MlbTeam.java
package com.gelakinetic.mlb_elo_calculator;

import java.util.Comparator;

public class MlbTeam implements Comparable<MlbTeam> {

    String name;
    int wins;
    int losses;
    double elo;
    public int winLossRank;
    public int eloRank;

    MlbTeam(String n) {
        name = n;
        wins = 0;
        losses = 0;
        elo = EloCalculator.STARTING_ELO;
    }

    @Override
    public String toString() {
        return String.format("%-21s (%3d-%3d) ELO: %d", name, wins, losses, (int) Math.round(elo));
    }

    @Override
    public int compareTo(MlbTeam o) {
        return -1 * Double.compare(this.elo, o.elo);
    }

    static class MlbTeamWinsComparator implements Comparator<MlbTeam> {

        @Override
        public int compare(MlbTeam arg0, MlbTeam arg1) {
            return -1 * Integer.compare(arg0.wins, arg1.wins);
        }
    }

    static class MlbTeamRankChangeComparator implements Comparator<MlbTeam> {

        @Override
        public int compare(MlbTeam o1, MlbTeam o2) {
            return Integer.compare((o2.winLossRank - o2.eloRank), (o1.winLossRank - o1.eloRank));
        }

    }

}

File: src/com/gelakinetic/mlbEloCalculator/MlbTeam.kt
package com.gelakinetic.mlbEloCalculator

import java.util.Comparator
import kotlin.math.roundToInt

class MlbTeam internal constructor(internal var name: String) : Comparable<MlbTeam> {
    internal var wins = 0
    internal var losses = 0
    internal var elo = EloCalculator.STARTING_ELO
    internal var winLossRank = 0
    internal var eloRank = 0

    override fun toString(): String {
        return String.format("%-21s (%3d-%3d) ELO: %d", name, wins, losses, elo.roundToInt())
    }

    override fun compareTo(other: MlbTeam): Int {
        return -1 * this.elo.compareTo(other.elo)
    }

    internal class MlbTeamEloComparator : Comparator<MlbTeam> {
        override fun compare(arg0: MlbTeam, arg1: MlbTeam): Int {
            return -1 * arg0.elo.compareTo(arg1.elo)
        }
    }

    internal class MlbTeamWinsComparator : Comparator<MlbTeam> {
        override fun compare(arg0: MlbTeam, arg1: MlbTeam): Int {
            return -1 * arg0.wins.compareTo(arg1.wins)
        }
    }

    internal class MlbTeamRankChangeComparator : Comparator<MlbTeam> {
        override fun compare(o1: MlbTeam, o2: MlbTeam): Int {
            return (o2.winLossRank - o2.eloRank).compareTo(o1.winLossRank - o1.eloRank)
        }
    }
}